src/std/net/json-rpc.ss

```diff
@@ -126,22 +106,121 @@
   (def request-string (request->string method (param-encoder params) id))
   (when log
     (log [to: server-url request: request-string]))
-
   ;; TODO: implement timeouts, with semi-asynchronous aborts the http-post thread itself.
   (def response-bytes
     (request-response-bytes
      (http-post server-url
                 auth: auth
-                headers: `(("Content-Type" . "application/json") ,@headers)
+                headers: `(("Content-Type" . "application/json-rpc")
+                           ;; ("Accept" . "application/json-rpc, application/json, application/jsonrequest")
+                           ,@headers)
                 cookies: cookies
                 data: (string->bytes request-string))))
   (def response-json
     (try (content->json response-bytes) ;; todo: move to decode-json-rpc-response ?
          (catch (e) (raise (malformed-response request-id: id response: response-bytes e: e)))))
   (when log
     (log [from: server-url response: (bytes->string response-bytes)]))
-
   (decode-json-rpc-response
    result-decoder
    (and (hash-table? response-json) (hash-get response-json "id"))
    response-json))
+
+(def (request->string method params id)
+  (try (json-object->string
+        (json-rpc-request jsonrpc: json-rpc-version method: method params: params id: id))
+       (catch (e) (raise (malformed-request method: method params: params e: (error-message e))))))
+
+(def (decode-json-rpc-response decoder request-id response-json)
+  (def (mal e)
```

Comment:
better name please!

Reply:
`malformed-response!` or something.

Reply:
I'm not sure what you mean. It's only a condition object being thrown. In general, I long for a structured condition system much more like CL's.

Reply:
`malformed-response` is just the condition. `raise` (without `!`) is the function that cuts. I'll rename the `mal` helper to `mal!` though.